w2w/w2w.py

```diff
@@ -1183,107 +1250,104 @@ def expand_land_cat_parents(
+    )
     ifile = info['dst_lcz_params_file']
     da = xr.open_dataset(ifile)
     if not 'URB_PARAM' in list(da.keys()):
-        print(f"WARNING: URB_PARAM matrix not present")
+        print(f'WARNING: URB_PARAM matrix not present')
     else:
-        print(f"OK")
-    print("")
+        print(f'OK')
+    print('')
 
-    print(f"Check 5: FRC_URB2D present in "
-          f"{info['dst_lcz_params_file'].split('/')[-1]}?")
+    print(
+        f'Check 5: FRC_URB2D present in '
+        f"{info['dst_lcz_params_file'].split('/')[-1]}?"
+    )
     ifile = info['dst_lcz_params_file']
     da = xr.open_dataset(ifile)
     if not 'FRC_URB2D' in list(da.keys()):
-        print(f"WARNING: FRC_URB2D not present in {ifile}")
+        print(f'WARNING: FRC_URB2D not present in {ifile}')
     else:
         FRC_URB2D = da.FRC_URB2D.values
-        print(f"OK: FRC_URB2D values range between "
-              f"{'{:0.2f}'.format(FRC_URB2D.min())} and "
-              f"{'{:0.2f}'.format(FRC_URB2D.max())}")
-    print("")
+        print(
+            f'OK: FRC_URB2D values range between '
+            f"{'{:0.2f}'.format(FRC_URB2D.min())} and "
+            f"{'{:0.2f}'.format(FRC_URB2D.max())}"
+        )
+    print('')
 
-    print("Check 6: Do URB_PARAM variable values follow expected range in "
-          f"{info['dst_lcz_params_file'].split('/')[-1]}?")
+    print(
+        'Check 6: Do URB_PARAM variable values follow expected range in '
+        f"{info['dst_lcz_params_file'].split('/')[-1]}?"
+    )
     ifile = info['dst_lcz_params_file']
     da = xr.open_dataset(ifile)
 
     ucp_dict = {
-        'LP_URB2D'  : {
-            'index': 91,
-            'range': [0,1]
-        },
-        'MH_URB2D'  : {
+        'LP_URB2D': {'index': 91, 'range': [0, 1]},
+        'MH_URB2D': {
             'index': 92,
-            'range': [0, ucp_table['MH_URB2D'].max() + ucp_table['MH_URB2D'].std()]
+            'range': [0, ucp_table['MH_URB2D'].max() + ucp_table['MH_URB2D'].std()],
         },
-        'HGT_URB2D' : {
+        'HGT_URB2D': {
             'index': 94,
-            'range': [0, ucp_table['MH_URB2D'].max() + ucp_table['MH_URB2D'].std()]
-        },
-        'LB_URB2D'  : {
-            'index': 95,
-            'range': [0, 5]
-        },
-        'LF_URB2D'  : {
-            'index': 96,
-            'range': [0, 5]
-        },
-        'LF_URB2D'  : {
-            'index': 97,
-            'range': [0, 5]
-        },
-        'LF_URB2D'  : {
-            'index': 98,
-            'range': [0, 5]
-        },
-        'LF_URB2D'  : {
-            'index': 99,
-            'range': [0, 5]
+            'range': [0, ucp_table['MH_URB2D'].max() + ucp_table['MH_URB2D'].std()],
         },
+        'LB_URB2D': {'index': 95, 'range': [0, 5]},
+        'LF_URB2D': {'index': 96, 'range': [0, 5]},
+        'LF_URB2D': {'index': 97, 'range': [0, 5]},
+        'LF_URB2D': {'index': 98, 'range': [0, 5]},
+        'LF_URB2D': {'index': 99, 'range': [0, 5]},
```

Comment:
mhm, I still don't fully understand why we assign different values to the same key -- simply overriding them and leaving it at `{'index': 99, 'range': [0, 5]}` in the end...

Reply:
w2w adds a 3D URB_PARAMS matrix to the geo_em file. This matrix stores all urban parameters, and LF_URD2D for four directions are stored in positions 96-99 (even though with same values). 

So that is what the index is referring to. See als [this document](https://ral.ucar.edu/sites/default/files/public/product-tool/NUDAPT_44_Documentation.pdf), Table on pages 5 and 6.

In this section of the code, the 'LF_URB2D' keys could get an additional subscript _1, _2, ... the thing that is important is that the values at index 96, 97, .... are checked.